calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -195,35 +195,49 @@ func (rh resourceHelper) Update(ctx context.Context, client client.Interface, re
 	// Check to see if the resourceVersion is specified in the resource object.
 	rv := resource.(ResourceObject).GetObjectMeta().GetResourceVersion()
 
+	// Copy the resource to prevent modifying the input resource metadata.
+	ru := resource.DeepCopyObject().(ResourceObject)
+
 	// If the resourceVersion is specified then we use it to try and update the resource.
 	// Do not attempt to retry if the resource version is specified.
 	if rv != "" {
+		// Clean out the resource version to always get the latest revision.
+		ru.(ResourceObject).GetObjectMeta().SetResourceVersion("")
 		// Validate the metadata is not changed for the the resource.
-		ro, err := rh.get(ctx, client, resource)
+		ro, err := rh.get(ctx, client, ru)
 		if err != nil {
 			return ro, err
 		}
-		// Set the specific resource version we are attempting to update.
-		ro.GetObjectMeta().SetResourceVersion(rv)
-		resource = mergeMetadataForUpdate(ro, resource)
+		// Check that the resource version is the latest
+		if rv != ro.GetObjectMeta().GetResourceVersion() {
+			id := ro.GetObjectMeta().GetName()
+			if ro.GetObjectMeta().GetNamespace() != "" {
+				id = fmt.Sprintf("%s/%s", ro.GetObjectMeta().GetNamespace(), ro.GetObjectMeta().GetName())
```

Comment:
sorry just spotted this:  I think this should also include the resource kind in it, so of the format `NetworkPolicy(namespace/name)`

Reply:
Currently the error message ends up looking like this:
`Failed to apply 'IPPool' resource: update conflict: default-ipv4-ippool`
If it already lists the resource type, should we still make the changes so it looks like:
`Failed to apply 'IPPool' resource: update conflict: IPPool(default-ipv4-ippool)`